django_sendfile/utils.py

```diff
@@ -81,23 +92,29 @@ def sendfile(request, filename, attachment=False, attachment_filename=None,
         ``False``: No ``Content-Disposition`` filename
         ``String``: Value used as filename
 
-    If neither ``mimetype`` or ``encoding`` are specified, then they will be guessed via the
-    filename (using the standard Python mimetypes module)
+    Any of ``encoding``, ``filesize and ``mimetype`` left to _guess will be guessed via the
+    filename (using the standard python mimetypes`and os.path modules).
+
+    Any of ``encoding``, ``filesize`` and ``mimetype`` set to None will not be set into the
+    response headers.
```

Comment:
The API is a bit confusing here.

On the one hand, setting `encoding`, `filezie`, and `mimetype` to a certain value will control if those headers are sent or not. On the other, controlling if the file is checked (i.e. required to be readable by the Django app) is done via `check_exist`.

I think we should choose one method or the other, not both.